test/cpp/test_aten_xla_tensor.cpp

```diff
@@ -5958,33 +5958,45 @@ TEST_F(AtenXlaTensorTest, TestNllLoss) {
   int batch = 6;
   int classes = 2;
   for (auto dtype : {torch::kFloat, torch::kDouble}) {
-    for (int ignore_index : {-1, 1, 5}) {
-      torch::Tensor input =
-          torch::rand({batch, classes}, torch::TensorOptions(dtype));
-      torch::Tensor target =
-          torch::randint(std::min(ignore_index, 0), classes, {batch},
-                         torch::TensorOptions(torch::kLong));
-      torch::Tensor undef_weight;
-      for (torch::Reduction::Reduction reduction :
-           {torch::Reduction::Mean, torch::Reduction::Sum}) {
-        torch::Tensor output =
-            torch::nll_loss(/*self=*/input, /*target=*/target,
-                            /*weight=*/undef_weight,
-                            /*reduction=*/reduction,
-                            /*ignore_index=*/ignore_index);
+    for (int ignore_index : {-1, 0, 1, 5}) {
+      for (bool def_weight : {false, true}) {
+        torch::Tensor input =
+            torch::rand({batch, classes}, torch::TensorOptions(dtype));
+        torch::Tensor target =
+            torch::randint(std::min(ignore_index, 0), classes, {batch},
+                           torch::TensorOptions(torch::kLong));
+        torch::Tensor weight;
+        if (def_weight) {
+          weight = torch::rand({classes}, torch::TensorOptions(dtype));
+        }
+        for (torch::Reduction::Reduction reduction :
+             {torch::Reduction::Mean, torch::Reduction::Sum,
+              torch::Reduction::None}) {
+          torch::Tensor output =
+              torch::nll_loss(/*self=*/input, /*target=*/target,
+                              /*weight=*/weight,
+                              /*reduction=*/reduction,
+                              /*ignore_index=*/ignore_index);
 
-        ForEachDevice([&](const torch::Device& device) {
-          torch::Tensor xla_input = CopyToDevice(input, device);
-          torch::Tensor xla_target = CopyToDevice(target, device);
-          torch::Tensor xla_output = torch::nll_loss(
-              /*self=*/xla_input, /*target=*/xla_target,
-              /*weight=*/undef_weight,
-              /*reduction=*/reduction, /*ignore_index=*/ignore_index);
-          AllClose(output, xla_output);
-        });
+          ForEachDevice([&](const torch::Device& device) {
+            torch::Tensor xla_input = CopyToDevice(input, device);
+            torch::Tensor xla_target = CopyToDevice(target, device);
+            torch::Tensor xla_weight =
+                def_weight ? CopyToDevice(weight, device) : torch::Tensor();
+            torch::Tensor xla_output = torch::nll_loss(
+                /*self=*/xla_input, /*target=*/xla_target,
+                /*weight=*/xla_weight,
+                /*reduction=*/reduction, /*ignore_index=*/ignore_index);
+            AllClose(output, xla_output);
+          });
+        }
       }
     }
   }
+
+  ExpectCounterNotChanged("aten::(?!_local_scalar_dense).*",
```

Comment:
Hmmm is part of test calling into `item()`?

Reply:
Hmm, IIRC it did. But maybe I am confusing it with the backward test.
Let me double check ...